core/roboconf-messaging-rabbitmq/src/main/java/net/roboconf/messaging/rabbitmq/RabbitMqClientFactory.java

```diff
@@ -0,0 +1,134 @@
+
+import net.roboconf.core.utils.Utils;
+import net.roboconf.messaging.MessagingConstants;
+import net.roboconf.messaging.client.IAgentClient;
+import net.roboconf.messaging.client.IDmClient;
+import net.roboconf.messaging.factory.MessagingClientFactory;
+import org.apache.felix.ipojo.annotations.Component;
+import org.apache.felix.ipojo.annotations.Instantiate;
+import org.apache.felix.ipojo.annotations.Property;
+import org.apache.felix.ipojo.annotations.Provides;
+import org.apache.felix.ipojo.annotations.StaticServiceProperty;
+import org.apache.felix.ipojo.annotations.Updated;
+
+/**
+ * Messaging client factory for Rabbit MQ.
+ *
+ * @author Pierre Bourret - Université Joseph Fourier
+ */
+@Component(name = "roboconf-messaging-client-factory-rabbitmq", publicFactory = false,
+		managedservice = "net.roboconf.messaging.rabbitmq")
+@Provides(specifications = MessagingClientFactory.class,
+		properties = @StaticServiceProperty(name = MessagingClientFactory.MESSAGING_TYPE_PROPERTY, type = "string",
+				value = MessagingConstants.FACTORY_RABBIT_MQ))
+@Instantiate(name = "Roboconf RabbitMQ Messaging Client Factory")
+public class RabbitMqClientFactory implements MessagingClientFactory {
+
+	// The connection properties.
+	private String messageServerIp;
+	private String messageServerUsername;
+	private String messageServerPassword;
+
+	// The created clients.
+	// References to the clients are *weak*, so we never prevent their garbage collection.
+	private final Set<RabbitMqClient> clients = Collections.newSetFromMap(new WeakHashMap<RabbitMqClient, Boolean>());
+
+	// The logger
+	private final Logger logger = Logger.getLogger(this.getClass().getName());
+
+	@Property(name = MessagingConstants.RABBITMQ_SERVER_IP)
+	public void setMessageServerIp( final String messageServerIp ) {
+		this.messageServerIp = messageServerIp;
+		this.logger.finer("Server IP set to " + messageServerIp);
+	}
+
+	@Property(name = MessagingConstants.RABBITMQ_SERVER_USERNAME)
+	public void setMessageServerUsername( final String messageServerUsername ) {
+		this.messageServerUsername = messageServerUsername;
+		this.logger.finer("Server username set to " + messageServerUsername);
+	}
+
+	@Property(name = MessagingConstants.RABBITMQ_SERVER_PASSWORD)
+	public void setMessageServerPassword( final String messageServerPassword ) {
+		this.messageServerPassword = messageServerPassword;
+		this.logger.finer("Server password set to " + messageServerPassword);
+	}
+
+	@Updated
+	public void reconfigure() {
+		// Set the properties for all created clients.
+
+		// Make fresh snapshots of the clients, as we don't want to reconfigure them while holding the lock.
+		final ArrayList<RabbitMqClient> clients;
+		synchronized (this) {
+			clients = new ArrayList<>(this.clients);
+		}
+
+		// Now reconfigure all the clients.
+		for (RabbitMqClient client : clients) {
+			try {
+				client.setParameters(this.messageServerIp, this.messageServerUsername, this.messageServerPassword);
```

Comment:
I do not understand this.
setParameters will update the internal properties, but it will not create new clients with the new configuration. In fact, we should "ask" the reconfigurable clients to create a new client for this factory.